packages/rehype-dom-parse/lib/index.js

```diff
@@ -28,22 +28,23 @@ export default function parse(options) {
   }
 }
 
+const DOCUMENT_FRAGMENT_NODE = 11
+
 /**
  * @param {string} htmlString
  * @returns {DocumentFragment}
  */
 function createFragment(htmlString) {
-  const fragment = document.createDocumentFragment()
-  const temporary = document.createElement('body')
-  temporary.innerHTML = htmlString
-  let child = temporary.firstChild
-
-  while (child) {
-    fragment.append(child)
-    child = temporary.firstChild
-  }
-
-  return fragment
+  const doc = createDocument(`<body>${htmlString}</body>`)
```

Comment:
```suggestion
  const doc = createDocument('<!doctype html><body>' + htmlString)
```

Just to make sure that it’s definitely seen as HTML5 and not as quirksmode or so.
And the `</body>` is unneeded here, so just the prefix does the same!